aws/resource_aws_lex_bot.go

```diff
@@ -0,0 +1,475 @@
+			Required:     true,
+			ValidateFunc: validation.IntBetween(1, 5),
+		},
+		"message": {
+			Type:     schema.TypeSet,
+			Required: true,
+			MinItems: 1,
+			MaxItems: 15,
+			Elem:     lexMessageResource,
+		},
+		"response_card": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringLenBetween(1, 50000),
+		},
+	},
+}
+
+func flattenLexPrompt(prompt *lexmodelbuildingservice.Prompt) (flattened map[string]interface{}) {
+	flattened = map[string]interface{}{}
+	flattened["max_attempts"] = aws.Int64Value(prompt.MaxAttempts)
+	flattened["message"] = flattenLexMessages(prompt.Messages)
+
+	if prompt.ResponseCard != nil {
+		flattened["response_card"] = aws.StringValue(prompt.ResponseCard)
+	}
+
+	return
+}
+
+func expandLexPrompt(rawObject interface{}) (prompt *lexmodelbuildingservice.Prompt) {
+	m := rawObject.([]interface{})[0].(map[string]interface{})
+
+	prompt = &lexmodelbuildingservice.Prompt{}
+	prompt.MaxAttempts = aws.Int64(int64(m["max_attempts"].(int)))
+	prompt.Messages = expandLexMessages(m["message"].(*schema.Set).List())
+
+	if v, ok := m["response_card"]; ok && v != "" {
+		prompt.ResponseCard = aws.String(v.(string))
+	}
+
+	return
+}
+
+func flattenLexIntents(intents []*lexmodelbuildingservice.Intent) (flattenedIntents []map[string]interface{}) {
+	for _, intent := range intents {
+		flattenedIntents = append(flattenedIntents, map[string]interface{}{
+			"intent_name":    aws.StringValue(intent.IntentName),
+			"intent_version": aws.StringValue(intent.IntentVersion),
+		})
+	}
+
+	return
+}
+
+// Expects a slice of maps representing the Lex objects.
+// The value passed into this function should have been run through the expandLexSet function.
+// Example: []map[intent_name: OrderFlowers intent_version: $LATEST]
+func expandLexIntents(rawValues []interface{}) []*lexmodelbuildingservice.Intent {
+	intents := make([]*lexmodelbuildingservice.Intent, 0, len(rawValues))
+
+	for _, rawValue := range rawValues {
+		value, ok := rawValue.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		intents = append(intents, &lexmodelbuildingservice.Intent{
+			IntentName:    aws.String(value["intent_name"].(string)),
+			IntentVersion: aws.String(value["intent_version"].(string)),
+		})
+	}
+
+	return intents
+}
+
+func resourceAwsLexBot() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLexBotCreate,
+		Read:   resourceAwsLexBotRead,
+		Update: resourceAwsLexBotUpdate,
+		Delete: resourceAwsLexBotDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+				if _, ok := d.GetOk("create_version"); !ok {
+					d.Set("create_version", true)
+				}
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Update: schema.DefaultTimeout(time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
```

Comment:
Since you originally created this PR, it looks like AWS has added a few new fields, like `DetectSentiment` and `EnableModelImprovements`. We should add support for these or create an Issue to add it later

Reply:
Yep. Also, added nluIntentConfidenceThreshold